src/librustc_mir/transform/qualify_consts.rs

```diff
@@ -1163,8 +1161,23 @@ impl MirPass for QualifyAndPromoteConstants {
             let (temps, candidates) = {
                 let mut qualifier = Qualifier::new(tcx, def_id, mir, mode);
                 if mode == Mode::ConstFn {
-                    // Enforce a constant-like CFG for `const fn`.
-                    qualifier.qualify_const();
+                    if tcx.is_min_const_fn(def_id) {
+                        // enforce `min_const_fn` for stable const fns
+                        use super::qualify_min_const_fn::is_min_const_fn;
+                        if let Err((span, err)) = is_min_const_fn(tcx, def_id, mir) {
+                            tcx.sess.span_err(span, &err);
+                        } else {
+                            // this should not produce any errors, but better safe than sorry
+                            // FIXME(oli-obk): still needed for drop/cell analysis, write an
+                            // analysis checking constants for drop/cell in the final value instead
+                            // of analyzing the bodies of initializers and functions
+                            // that should allow us to nuke most of this file.
```

Comment:
This is wrong, IMO, can you remove this comment and/or move it to an issue for further discussion?